src/robotFactory.js

```diff
@@ -1,10 +1,79 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: x,
+      y: y,
+    };
+  }
```

Comment:
Тут бы можно было упростить запись (Первый пункт чеклиста)
```suggestion
    this.coords = {
      x,
      y,
    };
  }
```

Reply:
Yeap, should fix this comment